src/client.rs

```diff
@@ -820,19 +820,18 @@ where
 
                 // The message will be forwarded to the server intact. We still would like to
                 // parse it below to figure out what to do with it.
-                let original = message.clone();
 
-                let code = message.get_u8() as char;
-                let _len = message.get_i32() as usize;
+                // Safe to unwrap because we know this message has a certain length and has the code
+                let code = *message.get(0).unwrap() as char;
```

Comment:
Maybe add a comment here
// Read the first byte of the message without advancing the internal pointer.

Reply:
Maybe change this to `unwrap_or("$")` , in the unlikely event that this branch is hit, it will not panic, instead it will hit the unsupported message branch in the `match` block